packages/components/tests/integration/components/hds/tabs/index-test.js

```diff
@@ -96,4 +98,94 @@ module('Integration | Component | hds/tabs/index', function (hooks) {
+        <T.Tab data-test="clicked tab">Two</T.Tab>
+        <T.Panel data-test="NOT clicked tab panel">Content 1</T.Panel>
+        <T.Panel data-test="clicked tab panel">Content 2</T.Panel>
+      </Hds::Tabs>
+    `);
+    await click('[data-test="clicked tab"] .hds-tabs__tab-button');
+    assert
+      .dom('[data-test="clicked tab"]')
+      .hasClass('hds-tabs__tab--is-selected');
+    assert
+      .dom('[data-test="clicked tab"] .hds-tabs__tab-button')
+      .hasAttribute('aria-selected');
+    assert
+      .dom('[data-test="clicked tab panel"]')
+      .doesNotHaveAttribute('hidden');
+
+    assert
+      .dom('[data-test="NOT clicked tab"]')
+      .doesNotHaveClass('hds-tabs__tab--is-selected');
+    assert
+      .dom('[data-test="NOT clicked tab"] .hds-tabs__tab-button')
+      .doesNotHaveAttribute('aria-selected');
+    assert.dom('[data-test="NOT clicked tab panel"]').hasAttribute('hidden');
+  });
+
+  // Test keyboard controls:
+
+  test('it should focus tabs and navigate through them using left and right arrow keys', async function (assert) {
+    const leftArrowKey = 37;
+    const rightArrowKey = 39;
+
+    await render(hbs`
+      <Hds::Tabs as |T|>
+        <T.Tab data-test="first tab">One</T.Tab>
+        <T.Tab data-test="secound tab">Two</T.Tab>
+        <T.Panel data-test="first panel">Content 1</T.Panel>
+        <T.Panel data-test="secound panel">Content 2</T.Panel>
+      </Hds::Tabs>
+    `);
+    // focus 2nd tab:
+    await focus('[data-test="secound tab"] .hds-tabs__tab-button');
+    // navigate to the previous (1st) tab using right arrow key:
+    await triggerKeyEvent(
+      '[data-test="secound tab"] .hds-tabs__tab-button',
+      'keyup',
+      rightArrowKey
+    );
+    // test that the navigated to tab is now focused:
+    assert.dom('[data-test="first tab"] .hds-tabs__tab-button').isFocused();
+
+    // navigate back to the next (2nd) tab using left arrow key:
+    await triggerKeyEvent(
+      '[data-test="first tab"] .hds-tabs__tab-button',
+      'keyup',
+      leftArrowKey
+    );
+    // test that the navigated to tab is now focused:
+    assert.dom('[data-test="secound tab"] .hds-tabs__tab-button').isFocused();
+  });
+
+  // Test Tab options:
+
+  test('it should render an icon when passed into a tab', async function (assert) {
+    await render(hbs`
+      <Hds::Tabs as |T|>
+        <T.Tab @icon="info">One</T.Tab>
+        <T.Tab>Two</T.Tab>
+        <T.Panel>Content 1</T.Panel>
+        <T.Panel>Content 2</T.Panel>
+      </Hds::Tabs>
+    `);
+    assert.dom('.hds-tabs__tab-icon').exists();
+  });
+
+  test('it should render a count when passed into a tab', async function (assert) {
+    await render(hbs`
+      <Hds::Tabs as |T|>
+        <T.Tab @count="5">One</T.Tab>
+        <T.Tab>Two</T.Tab>
+        <T.Panel>Content 1</T.Panel>
+        <T.Panel>Content 2</T.Panel>
+      </Hds::Tabs>
+    `);
+    assert.dom('.hds-tabs__tab-count').exists();
```

Comment:
probably I would also add test that checks that the correct text value is rendered:
```suggestion
    assert.dom('.hds-tabs__tab-count').exists();
    assert.dom('.hds-tabs__tab-count').hasText('5');
```